net/golioth/coap_req.c

```diff
@@ -731,3 +750,47 @@ int64_t golioth_coap_reqs_poll_prepare(struct golioth_client *client, int64_t no
 
 	return timeout;
 }
+
+static void golioth_coap_reqs_cancel_all_with_reason(struct golioth_client *client,
+						     int reason)
+{
+	struct golioth_coap_req *req, *next;
+
+	SYS_DLIST_FOR_EACH_CONTAINER_SAFE(&client->coap_reqs, req, next, node) {
+		struct golioth_req_rsp rsp = {
+			.user_data = req->user_data,
+			.err = reason,
+		};
+
+		(void)req->cb(&rsp);
```

Comment:
Should we check that the callback is not `NULL` first?

Reply:
No, we don't. `req->cb` is always initialized to some value. If user passes `NULL`, then `golioth_req_rsp_default_handler` is set in `golioth_coap_req_init()`.